pkg/sdkserver/server.go

```diff
@@ -0,0 +1,93 @@
+package sdkserver
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/acorn-io/broadcaster"
+	"github.com/gptscript-ai/gptscript/pkg/gptscript"
+	"github.com/gptscript-ai/gptscript/pkg/mvl"
+	gserver "github.com/gptscript-ai/gptscript/pkg/server"
+	"github.com/rs/cors"
+)
+
+type Options struct {
+	gptscript.Options
+
+	ListenAddress string
+	Debug         bool
+}
+
+func Start(ctx context.Context, opts Options) error {
+	sigCtx, cancel := signal.NotifyContext(ctx, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT, syscall.SIGKILL)
+	defer cancel()
+	go func() {
+		// This is a hack. This server will be run as a forked process in the SDKs. The SDKs will hold stdin open for as long
+		// as it wants the server running. When stdin is closed (or the parent process dies), then this will unblock and the
+		// server will be shutdown.
+		_, _ = io.ReadAll(os.Stdin)
+		cancel()
+	}()
+
+	if opts.Debug {
+		mvl.SetDebug()
+	}
+
+	events := broadcaster.New[gserver.Event]()
+	opts.Options.Runner.MonitorFactory = gserver.NewSessionFactory(events)
+	go events.Start(ctx)
+
+	g, err := gptscript.New(&opts.Options)
+	if err != nil {
+		return err
+	}
+
+	s := &server{
+		client: g,
+		events: events,
+	}
+	defer s.Close()
+
+	s.addRoutes(http.DefaultServeMux)
```

Comment:
Where does this `s` server end up getting used?

Reply:
In `addRoutes`, it sets up the routes for the various commands, which is where the client and events get used.